programs/zstdcli.c

```diff
@@ -1358,7 +1361,17 @@ int main(int argCount, const char* argv[])
 #endif
 
     /* No input filename ==> use stdin and stdout */
-    if (filenames->tableSize == 0) UTIL_refFilename(filenames, stdinmark);
+    if (filenames->tableSize == 0) {
+      /* It is possible that the input
+       was a number of empty directories. In this case
+       stdin and stdout should not be used */
+       if (nbInputFileNames > 0 ){
+        DISPLAYLEVEL(2, "please provide correct input file(s) or non-empty directories -- ignored \n");
+        CLEAN_RETURN(2);
```

Comment:
This should also be `CLEAN_RETURN(1);`

Reply:
Actually, if we want the CLI to return without erroring in this scenario, it should be `CLEAN_RETURN(0)`.